opentelemetry/proto/common/v1/common.proto

```diff
@@ -54,8 +54,6 @@ message ArrayValue {
 message KeyValueList {
   // A collection of key/value pairs of key-value pairs. The list may be empty (may
   // contain 0 elements).
-  // The keys MUST be unique (it is not allowed to have more than one
```

Comment:
On pdata (the library that all Collector components use, currently on v1.3.0) we assume that keys are unique and represent `KeyValueList`s with the [`pcommon.Map`](https://pkg.go.dev/go.opentelemetry.io/collector/pdata/pcommon#Map) type. This method has two methods that assume keys are unique: [`AsRaw`](https://pkg.go.dev/go.opentelemetry.io/collector/pdata/pcommon#Map.AsRaw) and [`Get`](https://pkg.go.dev/go.opentelemetry.io/collector/pdata/pcommon#Map.Get). I am not sure how we (Collector SIG) would address this change in a non-breaking way.

Reply:
The `Put*` functions are expect to update an existing item with the key not add a new one.

Reply:
If I understand correctly only the last item with the same key will be preserved. Right?

We could change the comment to something like (inspired by https://datatracker.ietf.org/doc/html/rfc8259#section-4):

> The keys SHOULD be unique. Key/value collection whose keys are all unique
   is interoperable in the sense
 that all software receiving that collection will agree on
   the key-value mappings. When the key within a collection are not
   unique, the behavior of software that receives such an object is
   unpredictable. Some receivers report the last key/value pair
   only. Other receivers report an error or fail to parse the
   collection, and some receivers report all of the name/value pairs,
   including duplicates.

However, I think that this would not progress towards https://github.com/open-telemetry/opentelemetry-specification/issues/3931 and would only bring ambiguity.

Reply:
> We make no guarantees as to what happens in this case, since it is an invalid situation.

Are you able to define it (and make a guarantee) in a way that it would not break existing components?